packages/wallet/src/components/PaymentPointerCard.tsx

```diff
@@ -0,0 +1,169 @@
+      )}
+      {...props}
+    >
+      {children}
+    </button>
+  )
+})
+PaymentPointerCardButton.displayName = 'PaymentPointerCardButton'
+
+const paymentPointerSchema = z.object({
+  paymentPointer: z.string().min(3)
+})
+
+export const PaymentPointerCard = ({
+  paymentPointer
+}: PaymentPointerCardProps) => {
+  const [openDialog, closeDialog] = useDialog()
+  const [isEditing, setIsEditing] = useState(false)
+  const form = useZodForm({
+    schema: paymentPointerSchema,
+    defaultValues: {
+      paymentPointer: paymentPointer.url.split('/')[1]
+    }
+  })
+
+  const cardRef = useRef<HTMLDivElement>(null)
+
+  const handleSubmit = form.handleSubmit((data) => {
+    console.log(data)
+    setIsEditing(false)
+  })
+
+  const handleDeleteConfirmation = () => {
+    console.log('deletion confirmed')
+  }
+
+  useOnClickOutside(cardRef, () => setIsEditing(false))
+
+  return (
+    <div ref={cardRef} className={cardStyles}>
+      {/* IDEA:
+        We can show a tooltip when the user is hovering the `Edit` or `Delete` buttons. 
+      */}
+      {isEditing ? (
+        <IconButton
+          aria-label="delete payment pointer"
+          className="test h-7 w-7 text-red-400 transition-transform duration-150 hover:scale-[115%]"
+          onClick={() =>
+            openDialog(
+              <ConfirmationDialog
+                confirmText="Delete payment pointer"
+                onConfirm={handleDeleteConfirmation}
+                onClose={closeDialog}
+              />
+            )
+          }
+        >
+          <X stroke="currentColor" strokeWidth={3} />
+        </IconButton>
+      ) : (
+        <IconButton
+          onClick={() => {
+            setIsEditing(!isEditing)
+          }}
+          aria-label="edit payment pointer"
+          className="h-7 w-7 text-green-7 transition-transform duration-150 hover:scale-[115%] hover:text-green-3"
+        >
+          <PencilSquare />
+        </IconButton>
+      )}
+      <div className="flex flex-1 items-center justify-between space-x-2">
+        {isEditing ? (
+          <Form
+            className="w-full"
+            form={form}
+            onSubmit={handleSubmit}
+            stack="h"
+          >
+            <PaymentPointerInput
+              required
+              autoFocus
+              {...form.register('paymentPointer')}
+            />
+            <PaymentPointerCardButton
+              type="submit"
+              aria-label="update payment pointer"
+            >
+              Save
+            </PaymentPointerCardButton>
+          </Form>
+        ) : (
+          <>
+            <span className="px-1 font-medium">{paymentPointer.url}</span>
+            <PaymentPointerCardButton aria-label="view payment pointer">
+              View
+            </PaymentPointerCardButton>
+          </>
+        )}
+      </div>
+    </div>
+  )
+}
+
+const cardStyles = `
```

Comment:
Placed the card styles here because prettier was rearranging them, if they were placed inside the component.